custom_components/elasticsearch/es_gateway.py

```diff
@@ -2,301 +2,493 @@
-        use_api_key = api_key is not None
+    def _is_ignorable_error(transport_err) -> bool:
+        """Determine if a transport error is ignorable."""
 
-        args = {
-            "hosts": [url],
-            "serializer": get_serializer(),
-            "verify_certs": verify_certs,
-            "ssl_show_warn": verify_certs,
-            "ca_certs": ca_certs,
-            "timeout": timeout,
-        }
+        if isinstance(transport_err, TransportError7 | TransportError8):
+            return isinstance(transport_err.status_code, int) and transport_err.status_code <= 403
 
-        if use_basic_auth:
-            args["http_auth"] = (username, password)
+        return False
 
-        if use_api_key:
-            args["headers"] = {"Authorization": f"ApiKey {api_key}"}
+    def schedule_next_test(self) -> None:
+        """Schedule the next connection test."""
+        self._next_test = time.monotonic() + 30
 
-        return args
+    def should_test(self):
+        """Determine if a test should be run."""
+        return self._next_test is None or self._next_test <= time.monotonic()
+
+    async def spin(self) -> None:
+        """Spin the event loop."""
+        await asyncio.sleep(1)
+
+    async def _connection_monitor_task(self, single_test: bool = False):
+        """Perform tasks required for connection monitoring."""
+
+        # Connection monitor event loop
+        while True:
+            if not self.should_test():
+                await self.spin()
+                continue
+
+            self.schedule_next_test()
+
+            # This part runs every 30 seconds
+            self._logger.debug("Checking status of the connection to [%s].", self.gateway.url)
+
+            # Backup our current state to _previous and update our active state
+            self._previous = self._active
+
+            try:
+                self._active = await self.test()
+            except err as err:
+                self._logger.exception("Connection test to [%s] failed: %s", self.gateway.url, err)
+
+            self.schedule_next_test()
+
+            if self._active and self._previous is None:
+                self._logger.info("Connection to [%s] has been established.", self.gateway.url)
+            if self._active and not self._previous:
+                self._logger.info("Connection to [%s] has been reestablished.", self.gateway.url)
+            elif self._active:
+                self._logger.debug("Connection test to [%s] was successful.", self.gateway.url)
+            else:
+                self._logger.error("Connection to [%s] is currently inactive.", self.gateway.url)
+
+            if single_test:
+                break
+
+    async def test(self) -> bool:
+        """Perform a connection test."""
+
+        return await self._gateway.test()
+
+    def start(self, config_entry):
+        """Start the connection monitor."""
+        if not self._use_connection_monitor:
```

Comment:
This throws an error causing initialization to fail, because `ConnectionMonitor` does not have an attribute named `_use_connection_monitor`:
> Exception during component initialization: 'ConnectionMonitor' object has no attribute '_use_connection_monitor'